src/main/java/htsjdk/samtools/util/GZIIndex.java

```diff
@@ -258,66 +261,88 @@ public static final GZIIndex loadIndex(final Path indexPath) throws IOException
         if (indexPath == null) {
             throw new IllegalArgumentException("null input path");
         }
+        try (final ReadableByteChannel channel = Files.newByteChannel(indexPath)) {
+            return loadIndex(indexPath.toUri().toString(), channel);
+        }
+    }
+
+    /**
+     * Loads the index from the provided input stream.
+     *
+     * @param indexIn the input stream for the index to load.
+     *
+     * @return loaded index.
+     *
+     * @throws IOException if an I/O error occurs.
+     */
+    public static final GZIIndex loadIndex(final String source, final InputStream indexIn) throws IOException {
```

Comment:
mention that source is used for error messages in the doc and can be null

Reply:
this is silly, but can you put the InputStream as the first parameter since it's the important one

Reply:
Fixed the javadoc.

However, I left source as the first parameter for consistency with `BlockCompressedIndexedFastaSequenceFile`, `ReferenceSequenceFileFactory`, `VariantContext`, tribble's `IndexFactory`, etc.

Reply:
Ah, good point.